src/main/webapp/src/app/location/location.component.ts

```diff
@@ -0,0 +1,83 @@
+import { Component, Output, EventEmitter, ViewChild, ElementRef } from '@angular/core';
+import { LocationService } from '../location.service';
+import { Location } from '../location';
+import { string } from 'prop-types';
+
+@Component({
+  selector: 'app-location',
+  templateUrl: './location.component.html',
+  styleUrls: ['./location.component.css']
+})
+export class LocationComponent{
+
+  @Output() locationStored: EventEmitter<void> = new EventEmitter()
+	location = {} as Location;
+	locationFound = false;
+
+  @ViewChild("errorMessage", { static: false })
+  errorMessage!: ElementRef;
+
+  constructor(private locationService: LocationService) { }
+
+  handleLocationError(browserHasGeolocation: boolean, message: string): void {
+
+    browserHasGeolocation
+      ? this.errorMessage.nativeElement.innerText = "Geolocation services have failed. Try a default location. " + message
+      : this.errorMessage.nativeElement.innerText = "Geolocation is not supported by this browser. Try a default location."  
+
+  }
+
+  locationSuccess(position: Position): void {
+    this.location.lat = position.coords.latitude;
+    this.location.lng = position.coords.longitude;
+    this.setLocation();
+  }
+
+  locationError(error : PositionError): void {
+
+		var message:string;
+    message = "An unknown error occured/"
+    
+		switch(error.code) {
+			case error.PERMISSION_DENIED:
+				message = "User denied the request for Geolocation."
+				break;
+			case error.POSITION_UNAVAILABLE:
+				message = "Location information is unavailable."
+				break;
+			case error.TIMEOUT:
+				message = "The request to get user location timed out."
+				break;
+		}
+    this.handleLocationError(true, message);
+	}
+	
+  enableNavigatorLocation(): void {
+    if (navigator.geolocation) {
+      navigator.geolocation.getCurrentPosition(
+        this.locationSuccess = this.locationSuccess.bind(this),
+        this.locationError = this.locationError.bind(this),
+        {timeout:10000});
```

Comment:
is timeout here to stop it from loading if its been too long?

Reply:
Yes whilst I was testing I noticed that if Navigator.Geolocation work on the user's location, but the location services are turned off, it just hangs. So I added a 10 secs timeout, so the user would know to select a default option, because the geolocation couldn't find it's location. (It approximately takes me 4-5 seconds to run when location is turned on for my browser, so made it 10 secs?)

Reply:
always good to have a timeout on requests, maybe might be worth making this a constant in the class or file, maybe something like GEO_LOCATION_TIMEOUT_MS, that way you know what it is immediately and you know what units it's using.

Reply:
@lorikeet-s  I added the GEO_LOCATION_TIMEOUT_MS as a readonly. Not sure what the best way to declare this is. For some reason you can't define a const in a class in typescript?

Reply:
oh yes sorry, readonly is a good way of doing it, it does technically mean it can get reassigned in the constructor but otherwise it's constant.
Other options are put it above the class, then you can use const, and it's basically a global variables
or make it static readonly inside the class, then I believe it actually is const. 
But also this is fine, I'll leave it up to you if you want to change it.